janitor/functions.py

```diff
@@ -2630,3 +2630,51 @@ def to_datetime(df: pd.DataFrame, column: str, **kwargs) -> pd.DataFrame:
     df[column] = pd.to_datetime(df[column], **kwargs)
 
     return df
+
+
+@pf.register_dataframe_method
+def groupby_agg(
+        df: pd.DataFrame,
+        by: str,
+        new_column: str,
+        agg_column: str,
+        agg: Union[Callable, str, List, Dict],
+        axis: int=0
+) -> pd.DataFrame:
+    """
+
+    Allow one to chain a groupby and a merge
+
+    Functional usage example:
+
+    .. code-block:: python
```

Comment:
I'm trying to nudge people away from functional usage of pyjanitor and towards the method-chainable API, so perhaps we can kickstart the removal of the functional API examples (even though it's technically still possible) starting from here?